### ISSUE TITLE
[BUG] Can't launch executable with xfce4-terminal (Text ended before matching quote was found)

### ISSUE THREAD

**Issue by 1nikolas** [bottlesdevs/Bottles#1097]
**Describe the bug**
More bugs with the terminal integration (yay)
Executables with "launch with terminal" option don't run on xfce4-terminal (see log bellow). I suspect that it has to do with the file name/path having spaces.
If #1095 gets fixed I'll try for xterm and Konsole too

**Installation**
 - Method: aur
 - Version: upstream (bottles-git)

**To Reproduce**
1. Have xfce4-terminal
2. Create a bottle
3. Open an executable with "launch with terminal"
4. Neither terminal nor the executable runs

**Expected behavior**
The terminal and the executable runs

**Screenshots**
![image](https://user-images.githubusercontent.com/30593419/155891237-5dc63f36-dead-4352-ac36-e944b982a15f.png)

**Desktop (please complete the following information):**
 - OS: archlinux
 - Version: latest
 - Display server: Xorg

**Additional context**
Log:
```
(18:14:53) INFO Launching an executable��� 
(18:14:53) INFO Setting Key: [Latest_Executables] to [[{'name': 'Firefox Setup 97.0.1.exe', 'file': '/home/nick/Downloads/Firefox Setup 97.0.1.exe', 'args': ''}]] for bottle: [test]��� 
(18:14:53) WARNING No GPU vendor found, keep going without setting VK_ICD_FILENAMES.. 
xfce4-terminal: Text ended before matching quote was found for ". (The text was ?sh -c "/home/nick/.local/share/bottles/runners/caffe-7.2/bin/wine64 /home/nick/Downloads/Firefox?)
```

**Comment by mirkobrombin** [comment]
Can you attach the bottle.yml file?

**Comment by 1nikolas** [comment]
okay here it is

```
Arch: win64
Creation_Date: '2022-02-27 15:21:29.156351'
Custom_Path: false
DLL_Overrides: {}
DXVK: dxvk-1.9.4
Environment: Application
Environment_Variables: {}
External_Programs: {}
Installed_Dependencies: []
Latest_Executables:
-   args: ''
    file: /home/nick/Downloads/Firefox Setup 97.0.1.exe
    name: Firefox Setup 97.0.1.exe
NVAPI: dxvk-nvapi-v0.5.2
Name: test
Parameters:
    aco_compiler: false
    custom_dpi: 96
    discrete_gpu: false
    dxvk: true
    dxvk_hud: false
    dxvk_nvapi: false
    fixme_logs: false
    fsr: false
    fsr_level: 5
    fullscreen_capture: false
    gamemode: false
    gamescope: false
    gamescope_borderless: false
    gamescope_fps: 0
    gamescope_fps_no_focus: 0
    gamescope_fullscreen: true
    gamescope_game_height: 0
    gamescope_game_width: 0
    gamescope_scaling: false
    gamescope_window_height: 0
    gamescope_window_width: 0
    pulseaudio_latency: false
    sync: wine
    take_focus: false
    use_runtime: false
    virtual_desktop: false
    virtual_desktop_res: 1280x720
    vkd3d: true
Path: test
Runner: caffe-7.2
State: 0
Uninstallers: {}
Update_Date: '2022-02-27 15:21:29.156381'
VKD3D: vkd3d-proton-2.5
Versioning: false
Windows: win10
WorkingDir: ''
```

**Comment by mirkobrombin** [comment]
> > Can you attach the bottle.yml file?
> 
> I think this is an integration issue but sure I will when I get home

Ya, I was just checking for env vars and maybe other things which can cause these problems. So it is an implementation problem but we fixed this more then 2 times :| I'll check again

**Comment by 1nikolas** [comment]
well this commit introduces a new issue

![image](https://user-images.githubusercontent.com/30593419/157931262-07751d99-f43e-43e7-8e43-bcfd09949728.png)

**Comment by mirkobrombin** [comment]
Ok i got it

**Comment by 1nikolas** [comment]
Now it doesn't like the parenthesis in the program files folder

```
/bin/sh: -c: line 1: syntax error near unexpected token `('
/bin/sh: -c: line 1: `xfce4-terminal -e 'sh -c "/home/nick/.local/share/bottles/runners/caffe-7.2/bin/wine64 '/home/nick/.local/share/bottles/bottles/c-test/drive_c/Program Files (x86)/Program/program.exe'"''
```

**Comment by mirkobrombin** [comment]
> Now it doesn't like the parenthesis in the program files folder
> 
> ```
> /bin/sh: -c: line 1: syntax error near unexpected token `('
> /bin/sh: -c: line 1: `xfce4-terminal -e 'sh -c "/home/nick/.local/share/bottles/runners/caffe-7.2/bin/wine64 '/home/nick/.local/share/bottles/bottles/c-test/drive_c/Program Files (x86)/Program/program.exe'"''
> ```

And now I hate it :(

**Comment by mirkobrombin** [comment]
![image](https://user-images.githubusercontent.com/9479903/159312636-e502344e-de35-40b7-b118-5b77cfd95198.png)

Seem fixed now (xfce4-terminal).
